app/Tools/Features.php

```diff
@@ -20,16 +21,40 @@ public function __construct(array $config)
      */
     public function isEnabled($feature)
     {
-        if (isset($this->config[$feature])) {
-            if (!is_array($this->config[$feature])) {
-                return !!$this->config[$feature];
+        $config = $this->configRepo->get('features');
+
+        if (isset($config->$feature)) {
+            if (!is_array($config->$feature)) {
+                return !!$config->$feature;
             }
 
-            if (isset($this->config[$feature]['enabled'])) {
-                return !!$this->config[$feature]['enabled'];
+            if (isset($config->$feature['enabled'])) {
+                return !!$config->$feature['enabled'];
             }
         }
 
         return false;
     }
+
+    /**
+     * Get limit for feature
+     * @param  string $feature
+     * @return int|INF
+     */
+    public function getLimit($feature)
+    {
+        $config = $this->configRepo->get('features');
+
+        if (isset($config->limits[$feature])) {
+            if ($config->limits[$feature] === true) {
+                // Return infinity ie. unlimited
+                return INF;
+            }
+
+            return (int) $config->limits[$feature];
+        }
+
+        // Return infinity ie. unlimited
+        return INF;
```

Comment:
More of a question: I couldn't find another example where we return INF in the codebase, and though it makes total sense logically, I seem to remember that it can get treated differently with different interpreters. Of course, I can't find an example now, aside from HHVM, so maybe I'm writing this comment for no useful reason.

Reply:
I didn't even know it existed until I went looking for a better option than returning `true` and then have to test for it specially